crates/snp_validation/src/config.rs

```diff
@@ -0,0 +1,177 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use crate::{PlatformInfoFlags, PolicyFlags};
+use cryptography::digest::Digest;
+use flagset::FlagSet;
+use semver::VersionReq;
+use serde::{Deserialize, Deserializer, Serialize};
+
+#[derive(Clone, Deserialize, Debug, Default, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct ConfigIntermediate {
+    /// Values for `author_key_digest` in the report body.
+    /// This is the list of public keys which have signed the signing key of the Enarx binary.
+    #[serde(default)]
+    pub enarx_signer: Vec<Digest>,
+
+    /// Values for `measurement` in the report body.
+    /// This is the hash of the Enclave environment after the Enarx binary is loaded
+    /// but before any workload is loaded, so this is a hash of the Enarx binary
+    /// in memory.
+    #[serde(default)]
+    pub enarx_hash: Vec<Digest>,
+
+    /// Values for `id_key_digest` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    pub id_key_digest: Vec<Digest>,
+
+    /// Prohibited values for `id_key_digest` in the report body.
+    /// This is the list of public keys which have signed the Enarx binary.
+    #[serde(default)]
+    pub id_key_digest_blacklist: Vec<Digest>,
+
+    /// Minimum value for `policy.abi_major`.`policy.abi_minor`
+    #[serde(default)]
+    pub minimum_abi: VersionReq,
+
+    #[serde(default)]
+    #[serde(deserialize_with = "from_policy_string")]
+    pub policy_flags: Option<u8>,
```

Comment:
It really looks like there should be a `PolicyFlags` type

Reply:
Fixed.